src/prefect_server/api/runs.py

```diff
@@ -173,13 +182,20 @@ async def _create_flow_run(
     elif flow.archived:
         raise ValueError(f"Flow {flow.id} is archived.")
 
+    # determine active run_config
+    if run_config is None:
+        if flow.flow_group.run_config is not None:
+            run_config = flow.flow_group.run_config
+        else:
+            run_config = flow.run_config
+
     # set labels
     if labels is not None:
         run_labels = labels
     elif flow.flow_group.labels is not None:
         run_labels = flow.flow_group.labels
-    elif flow.run_config is not None:
-        run_labels = flow.run_config.get("labels") or []
+    elif run_config is not None:
```

Comment:
I think the logic in this section needs to be rearranged; specifically if a user provides a run config (with labels) but forgets to redundantly supply the labels in the API call, then the labels will be pulled from the flow group instead of the more recent / immediate run config object.  

One way we could achieve this is via an `else:` statement on the `if run_config is None:` clause above that does something like:
```python
elif labels is None:
    labels = run_config.get("labels")
```

and then this `elif` could be removed entirely.

Reply:
Hmmm, I thought about that, but decided the order I described above made more sense to me. Happy to swap though, I can see arguments for both ways.

Reply:
Swapped the resolution order (this wasn't a bug per-se, just a different idea of what was expected behavior).